modin/pandas/dataframe.py

```diff
@@ -2496,18 +2496,22 @@ def __setattr__(self, key, value):
         # - `_siblings`, which Modin initializes before it appears in __dict__
         if key in ["_query_compiler", "_siblings"] or key in self.__dict__:
             pass
-        elif key in self and key not in dir(self):
+        elif key not in dir(self) and key not in self and not is_list_like(value):
```

Comment:
This seems some what of confusing logic to me. I think the way it was expressed the comment above somewhat clearly: if we have an attribute that already exists (column that already exists), we can go ahead and assign the new value. However, if that's not the case, we should promptly let the user know that we can't set new columns as an attribute.